### PR TITLE
feat(streaming): support create mv on index

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

This PR fixes several bugs and fully support create mv on index.

* A separate struct to store arrangement info, so that lookup can work correctly.
* Special case for chain. Need to set first barrier's prev epoch to 0. See comments for more information.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

ref https://github.com/singularity-data/risingwave/issues/2016
may be blocked by https://github.com/singularity-data/risingwave/issues/2151

### CODE REVIEW

**Comment by skyzh** [comment]
Unit test because of a new constraint is added, and should be fixed quickly. I'll continue to work on this tomorrow.

File: src/stream/src/executor_v2/lookup.rs [hunk 1]
```diff
@@ -122,40 +123,30 @@ impl ExecutorBuilder for LookupExecutorBuilder {
...
-        );
-        let arrangement_keyspace_id =
-            unique_operator_id(lookup.arrange_fragment_id, lookup.arrange_operator_id);
+        let arrangement_table_id = match lookup.arrangement_table_id.as_ref().unwrap() {
+            ArrangementTableId::IndexId(x) => *x,
+            ArrangementTableId::TableId(x) => *x,
+        };
+
+        let arrangement_col_descs = lookup
+            .get_arrangement_table_info()?
```
Comment (yuhao-su) [inline src/stream/src/executor_v2/lookup.rs:139]: This is essentially from `order_desc` in `TableDesc`, which is arrange key + pk.
But the comment of `arrangement_col_descs` in `LookupExecutorParams` wrote
```
/// From the perspective of arrangements, `arrangement_col_descs` include all columns of the
/// `MaterializeExecutor`. For example, if we already have a table with 3 columns: `a, b,
/// _row_id`, and we create an arrangement with join key `a` on it. `arrangement_col_descs`
/// should contain all 3 columns.
```

Comment (skyzh) [reply src/stream/src/executor_v2/lookup.rs:139]: Nope, that's all columns 🤣 Index contains all columns, and its column desc also contains all columns.

File: proto/stream_plan.proto [hunk 2]
```diff
@@ -162,11 +166,20 @@ message BatchPlanNode {
   uint32 parallel_unit_id = 5;
 }
 
+message ArrangementInfo {
+  // Column indexes and orders of primary key
+  repeated plan_common.ColumnOrder column_orders = 1;
```
[the remark this answers is not in the document]
Comment (skyzh) [reply proto/stream_plan.proto:171]: I'm thinking of a term to describe the prefix of materialize executor. Let's name it as arrange key?

File: src/stream/src/executor_v2/lookup.rs (hunk 1 above)
Comment (yuhao-su) [reply src/stream/src/executor_v2/lookup.rs:139]: Oh sry, my mistake.

**Review by yuhao-su — APPROVED** [review]

**Comment by yuhao-su** [comment]
LGTM!

**Pull request merged**